docs/sources/developers/plugins/add-authentication-for-data-source-plugins.md

```diff
@@ -0,0 +1,159 @@
+
+## Add a proxy route to your plugin
+
+To forward requests through the Grafana proxy, you need to configure one or more _routes_. A route specifies how the proxy will transform outgoing requests. All routes for a given plugin are defined in the [plugin.json]({{< relref "metadata.md" >}}) file.
+
+Let's add a route to proxy requests to `https://api.example.com/foo/bar`.
+
+1. Add the route to `plugin.json`. When you make a request
+
+   ```json
+   "routes": [
+     {
+       "path": "example",
+       "url": "https://api.example.com"
+     }
+   ]
+   ```
+
+1. In the `DataSource`, extract the proxy URL from `instanceSettings` to a class property called `url`.
+
+   ```ts
+   export class DataSource extends DataSourceApi<MyQuery, MyDataSourceOptions> {
+     url?: string;
+
+     constructor(instanceSettings: DataSourceInstanceSettings<MyDataSourceOptions>) {
+       super(instanceSettings);
+
+       this.url = instanceSettings.url;
+     }
+
+     // ...
+
+   }
+   ```
+
+1. In the `query` method, make a request using [BackendSrv]({{< relref "../../packages_api/runtime/backendsrv.md" >}}).
+
+   ```ts
+   const routePath = '/example';
+
+   getBackendSrv()
+     .datasourceRequest({
+       url: this.url + routePath + '/foo/bar',
+       method: 'GET',
+     });
+   ```
+
+> You need to reload the Grafana server every time you make a change to your `plugin.json` file.
+
+## Add a dynamic proxy route to your plugin
+
+In the example above, the URL stays the same for anyone using the plugin. Let's look at how you can create dynamic routes based on data source options that are provided by the user.
+
+Many of the properties in the `route` object can use templates in the form of `{{ .JsonData.YOUR_OPTION_NAME }}`, where `YOU_OPTION_NAME` is the name of a property in the `jsonData` object.
+
+```json
+"routes": [
+  {
+    "path": "example",
+    "url": "https://api.example.com/projects/{{ .JsonData.projectId }}"
+  }
+]
+```
+
+You can also used secret options in your route configuration by changing `.JsonData` into `.SecureJsonData`.
+
+```json
+"routes": [
+  {
+    "path": "example",
+    "url": "https://{{ .JsonData.username }}:{{ .SecureJsonData.password }}@api.example.com"
+  }
+]
+```
+
+Now that you know how to define routes for your data source plugin. Next, let's look at how to authenticate requests for your routes.
+
+## Configure the authentication method for a route
+
+The Grafana proxy supports a number of different authentication methods. For more information on how to configure each authentication method, refer to [plugin.json]({{< relref "metadata.md" >}}).
+
+For any sensitive data, make sure that you [Encrypt data source options]({{< relref "encrypt-data-source-options.md" >}}), and that you use `{{ .SecureJsonData.YOUR_OPTION_NAME }}` when using secret data source options in your routes.
```

Comment:
Mixing secret and sensitive wordings in different places. Maybe stick with one, sensitive?